### PR TITLE
Document the declarative plugins configuration file

### PR DESCRIPTION
**Pull request by pugnascotia**
Follow-up to #77544.

### CODE REVIEW

File: docs/plugins/plugin-script.asciidoc [hunk 1]
```diff
@@ -279,3 +285,53 @@ The default location of the `plugins` directory depends on which package you ins
...
+. Docker only
+=====================
+At present, this feature is only available when running {es} using our
+Docker images. Other distributions of {es} will not start if they find a
+plugin configuration file.
+=====================
+
+You can choose to manage plugins using a declarative configuration file.
+When {es} starts up, it will compare the plugins in the file with those
+that are currently installed, and add or remove plugins as required. {es}
```
Comment (mark-vieira) [inline docs/plugins/plugin-script.asciidoc:304]: Should we clarify that only official plugins are automatically upgraded?

**Review by mark-vieira — APPROVED** [review]
One comment, otherwise LGTM.

File: docs/plugins/plugin-script.asciidoc [hunk 2]
```diff
@@ -279,3 +285,53 @@ The default location of the `plugins` directory depends on which package you ins
 * {ref}/zip-windows.html#windows-layout[Directory layout of Windows `.zip` archives]
 * {ref}/deb.html#deb-layout[Directory layout of Debian package]
 * {ref}/rpm.html#rpm-layout[Directory layout of RPM]
+
+
+[discrete]
```
Comment (jrodewig) [inline docs/plugins/plugin-script.asciidoc:290]: Right now, this content is appended to the [Plugins directory page](https://elasticsearch_80760.docs-preview.app.elstc.co/guide/en/elasticsearch/plugins/master/_plugins_directory.html), which seems odd to me. I'd drop the `[discrete]` tag so it renders on its own page.
```suggestion
```

File: docs/plugins/plugin-script.asciidoc [hunk 3]
```diff
@@ -279,3 +285,53 @@ The default location of the `plugins` directory depends on which package you ins
...
+[[manage-plugins-using-configuration-file]]
+=== Managing plugins using a configuration file
+
+[IMPORTANT]
+. Docker only
+=====================
+At present, this feature is only available when running {es} using our
+Docker images. Other distributions of {es} will not start if they find a
```
Comment (jrodewig) [inline docs/plugins/plugin-script.asciidoc:298]: I don't think we need "at present." Users can assume that.

I also wasn't sure what you meant by "our Docker images." If you mean the official ones, I'd link there so it's a bit more explicit.
```suggestion
This feature is only available for https://www.docker.elastic.co/[official {es}
Docker images]. Other {es} distributions will not start with a
```

File: docs/plugins/plugin-script.asciidoc [hunk 4]
```diff
@@ -279,3 +285,53 @@ The default location of the `plugins` directory depends on which package you ins
...
+plugin configuration file.
+=====================
+
+You can choose to manage plugins using a declarative configuration file.
```
Comment (jrodewig) [inline docs/plugins/plugin-script.asciidoc:302]:
```suggestion
If you run {es} using Docker, you can manage plugins using a declarative configuration file.
```

File: docs/plugins/plugin-script.asciidoc [hunk 5]
```diff
@@ -279,3 +285,53 @@ The default location of the `plugins` directory depends on which package you ins
...
+-----------------------------------
+plugins:
+  - id: analysis-icu
+  - id: repository-azure
+  - id: custom-mapper
+    location: https://example.com/archive/custom-mapper-1.0.0.zip
+-----------------------------------
+
+This example installs the official plugins `analysis-icu` and
+`repository-azure`, and one unofficial plugin. Every plugin must provide
```
Comment (jrodewig) [inline docs/plugins/plugin-script.asciidoc:321]:
```suggestion
This example installs the official `analysis-icu` and
`repository-azure` plugins, and one unofficial plugin. Every plugin must provide
```

File: docs/plugins/plugin-script.asciidoc [hunk 6]
```diff
@@ -279,3 +285,53 @@ The default location of the `plugins` directory depends on which package you ins
...
+an `id`, and unofficial plugins must also provide a `location`. This is
```
Comment (jrodewig) [inline docs/plugins/plugin-script.asciidoc:322]:
```suggestion
an `id`. Unofficial plugins must also provide a `location`. This is
```

File: docs/plugins/plugin-script.asciidoc [hunk 7]
```diff
@@ -279,3 +285,53 @@ The default location of the `plugins` directory depends on which package you ins
...
 * {ref}/rpm.html#rpm-layout[Directory layout of RPM]
+
+
+[discrete]
+[[manage-plugins-using-configuration-file]]
+=== Managing plugins using a configuration file
+
+[IMPORTANT]
+. Docker only
+=====================
```
Comment (jrodewig) [inline docs/plugins/plugin-script.asciidoc:296]: Not a big deal, but we only require four delimiters. No real reason to use more. For example:

```
[IMPORTANT]
.Docker only
====
My admon text
====
```

File: docs/plugins/plugin-script.asciidoc [hunk 8]
```diff
@@ -279,3 +285,53 @@ The default location of the `plugins` directory depends on which package you ins
...
+is an example:
+
+[source,yaml]
+-----------------------------------
+plugins:
+  - id: analysis-icu
+  - id: repository-azure
+  - id: custom-mapper
+    location: https://example.com/archive/custom-mapper-1.0.0.zip
+-----------------------------------
```
Comment (jrodewig) [inline docs/plugins/plugin-script.asciidoc:318]: Same comment re: delimiters as above.
```suggestion
----
plugins:
  - id: analysis-icu
  - id: repository-azure
  - id: custom-mapper
    location: https://example.com/archive/custom-mapper-1.0.0.zip
----
```

File: docs/plugins/plugin-script.asciidoc [hunk 9]
```diff
@@ -279,3 +285,53 @@ The default location of the `plugins` directory depends on which package you ins
...
+Java proxy system properties] when downloading plugins, you can also configure an
+HTTP proxy to use explicitly in the configuration file. For example:
+
+[source,yaml]
+-----------------------------------
+plugins:
+  - id: custom-mapper
+    location: https://example.com/archive/custom-mapper-1.0.0.zip
+proxy: proxy.example.com:8443
+-----------------------------------
```
Comment (jrodewig) [inline docs/plugins/plugin-script.asciidoc:337]: Minor delimiter fix.
```suggestion
[source,yaml]
----
plugins:
  - id: custom-mapper
    location: https://example.com/archive/custom-mapper-1.0.0.zip
proxy: proxy.example.com:8443
----
```

**Review by jrodewig — APPROVED** [review]
LGTM aside from some minor formatting fixes. The rest of my feedback consists of wording nits. Feel free to disregard those if wanted.

File: docs/plugins/plugin-script.asciidoc [hunk 10]
```diff
@@ -26,6 +26,12 @@ Otherwise run `bin/elasticsearch-plugin` as the user that owns all of the Elasti
 files.
 =====================
 
+=== Docker
```
[the remark this answers is not in the document]
Comment (pugnascotia) [reply docs/plugins/plugin-script.asciidoc:29]: It was intended but I'm happy to go with your suggestion.

File: docs/plugins/plugin-script.asciidoc (hunk 7 above)
Comment (pugnascotia) [reply docs/plugins/plugin-script.asciidoc:296]: I was copying from other lines in the file. I'll remember that for the future.

File: docs/plugins/plugin-script.asciidoc [hunk 11]
```diff
@@ -279,3 +285,53 @@ The default location of the `plugins` directory depends on which package you ins
...
 * {ref}/deb.html#deb-layout[Directory layout of Debian package]
 * {ref}/rpm.html#rpm-layout[Directory layout of RPM]
+
+
+[discrete]
+[[manage-plugins-using-configuration-file]]
+=== Managing plugins using a configuration file
+
+[IMPORTANT]
+. Docker only
```
[the remark this answers is not in the document]
Comment (pugnascotia) [reply docs/plugins/plugin-script.asciidoc:295]: I'd never have guess that 😁

**Comment by pugnascotia** [comment]
Thanks @jrodewig!

**Comment by pugnascotia** [comment]
Backports:

   * `7.16` -> 3be1cdfcf61
   * `8.0` -> 17a5b6a18f8

**Pull request merged**